message XmlUpdate{
	required string keyword = 1;
	required string value = 2;
}

message HandShake{
	required string headId = 1;
	required string bodyId = 2;
	required uint32 checksum = 3;
}

message ExternalConfig{
	required string messageid = 1;
	required uint32 targethost = 2;
	repeated XmlUpdate updateXml = 3;
	repeated string resetActivities = 4;
	required bool handOffRequest = 5;
}

message GenericACK{
	required uint32 hostid = 1;
	required string messageid = 2;
	optional HandShake handshaking = 3;
	required bool ownlock = 4;
}

message ResetMessage{
	repeated string resetActivities = 1;
}

message UpdateMessage{
	repeated XmlUpdate updateXml = 1;
}

